### PR TITLE
feat(skilavottord): update fix if new vehicle owner

### PR DESCRIPTION
**Pull request by essgje**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts [hunk 1]
```diff
@@ -49,6 +51,7 @@ export class VehicleService {
   }
 
   async findByVehicleId(vehicleId: string): Promise<VehicleModel> {
+    console.log('vehicle find by id...')
```
Comment (birkirkristmunds) [inline apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts:54]: Console.log ekki vinsælt

File: apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts [hunk 2]
```diff
@@ -60,20 +63,24 @@ export class VehicleService {
...
         }
+      } else {
+        await vehicle.save()
+        return true
       }
-      // Save vehicle to database
-      await vehicle.save()
-      return true
     } catch (err) {
+      this.logger.error('error creating vehicle', err)
```
Comment (birkirkristmunds) [inline apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts:83]: Mér var bent á að hafa loggerinn í template stíl um daginn. Sel það ekki dýrara en ég keypti það 

logger.error(
        `User ${user.nationalId} does not own the requested vehicle`,
        { permno: input.permno, user },
      )

**Review by birkirkristmunds — APPROVED** [review]

File: apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts (hunk 1 above)
Comment (essgje) [reply apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts:54]: gleymdi að taka út

File: apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts (hunk 2 above)
Comment (essgje) [reply apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts:83]: Ok en það er ekkert extra concat í gangi hér.

Comment (essgje) [reply apps/skilavottord/ws/src/app/modules/vehicle/vehicle.service.ts:83]: Skil ekki alveg :)

**Pull request merged**